lib/db/leveldb_dbinstance.go

```diff
@@ -217,19 +219,19 @@ func (db *Instance) updateFiles(folder, device []byte, fs []protocol.FileInfo, l
 		if err != nil {
 			panic(err)
 		}
-		// The Invalid flag might change without the version being bumped.
-		if !ef.Version.Equal(f.Version) || ef.Invalid != f.Invalid {
-			if isLocalDevice {
-				localSize.removeFile(ef)
-				localSize.addFile(f)
-			}
 
-			t.insertFile(folder, device, f)
-			if f.IsInvalid() {
-				t.removeFromGlobal(folder, device, name, globalSize)
-			} else {
-				t.updateGlobal(folder, device, f, globalSize)
-			}
+		// update the DB entry if the file has changed
```

Comment:
I had to remove an if statement here, because otherwise it would never update the record for the file and constantly rehash. This was due to the fact that only the checksum was changed. I could have added another check to the if statement, but it didn't seem to make sense. I did some testing and this function never gets called unless there was an actual change to the file.